src/python/bot/fuzzers/libfuzzer.py

```diff
@@ -883,6 +885,231 @@ def cleanse_crash(self,
       return result
 
 
+class AndroidLibFuzzerRunner(new_process.ProcessRunner, LibFuzzerCommon):
+  """Android libFuzzer runner."""
+  DEVICE_FUZZING_DIR = '/data/fuzz'
+
+  def __init__(self, executable_path, build_directory, default_args=None):
+    """Inits the AndroidLibFuzzerRunner.
+
+    Args:
+      executable_path: Path to the fuzzer executable.
+      build_directory: A MinijailChroot.
+      default_args: Default arguments to always pass to the fuzzer.
+    """
+    super(AndroidLibFuzzerRunner, self).__init__(
+        executable_path=android.adb.get_adb_path(),
+        default_args=self._get_default_args(executable_path, default_args))
+
+    self._copy_local_directory_to_device(build_directory)
+
+  def get_testcase_path(self, log_lines):
+    """Get testcase path from log lines."""
+    path = LibFuzzerCommon.get_testcase_path(self, log_lines)
+    if not path:
+      return path
+
+    return self._get_local_path(path)
+
+  def _get_default_args(self, executable_path, custom_default_args):
+    """Return a set of default arguments to pass to adb binary."""
+    default_args = ['shell']
+
+    # Add directory containing libclang_rt.ubsan_standalone-aarch64-android.so
+    # to LD_LIBRARY_PATH.
+    # FIXME: Remove this hardcode and get rid of this dependency completely.
+    default_args.append('LD_LIBRARY_PATH=/system/lib64:/system/lib64/vndk-R')
+
+    # Add sanitizer options.
+    default_args += environment.get_sanitizer_options_for_display()
+
+    default_args.append(self._get_device_path(executable_path))
+
+    if custom_default_args:
+      default_args += custom_default_args
+
+    return default_args
+
+  def _get_device_corpus_paths(self, corpus_directories):
+    """Return device paths for the given corpus directories."""
+    return [self._get_device_path(path) for path in corpus_directories]
+
+  def _get_device_path(self, path):
+    """Return device path for the given local path."""
+    root_directory = environment.get_root_directory()
+    return os.path.join(self.DEVICE_FUZZING_DIR,
+                        os.path.relpath(path, root_directory))
+
+  def _get_local_path(self, path):
+    """Return local path for the given device path."""
+    root_directory = environment.get_root_directory()
+    return os.path.join(root_directory,
+                        os.path.relpath(path, self.DEVICE_FUZZING_DIR))
```

Comment:
nit: maybe add something to check that the resulting path is a sane subdir of root_dir, in case e.g. the logs get corrupted somehow.

Reply:
Done!